proposals/0448-type-variable-scoping.rst

```diff
@@ -977,20 +1238,42 @@ Effects and Interactions
 The effects of this proposal are written out in the individual sections.
 Here, I summarize the effects on the principles_.
 
-1. We get closer to the `Lexical Scoping Principle`_:
-   with ``-Werror=pattern-signature-binds``, type variables cannot be bound in pattern signatures,
-   closing one of the places where the `Lexical Scoping Principle`_ is currently violated.
+#. The `Explicit Variable Principle`_ is made to hold, by allowing explicit binders for type variables for existentials and the variables bound by an inner ``forall`` in a higher-rank
+   type.
+
+#. The `Lexical Scoping Principle`_ is made to hold with ``-XNoImplicitBinds``
+
+   Indeed, the purpose of ``-XNoImplicitBinds`` is to be the single extension which is both necessary and sufficient to do this.
```

Comment:
Well, not completely: the Template Haskell objection still holds I think.

Reply:
```suggestion

#. The `Lexical Scoping Principle`_ outside of Template Haskell is made to hold with ``-XNoImplicitBinds``

   Indeed, the purpose of ``-XNoImplicitBinds`` is to be the single extension which is both necessary and sufficient to do this.
   (The Template Haskell issue is something would be solved by more rigorous notations of hygiene. That has little to do with ``-XNoImplicitBinds`` as currently specified, and is much more work of a very different nature.) 
```